cinema/serializers.py

```diff
@@ -68,13 +78,75 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available",
         )
 
+    @staticmethod
+    def get_tickets_available(obj):
+        sold_tickets_count = Ticket.objects.filter(movie_session=obj).count()
+        available_tickets = obj.cinema_hall.capacity - sold_tickets_count
+        return available_tickets
+
 
 class MovieSessionDetailSerializer(MovieSessionSerializer):
-    movie = MovieListSerializer(many=False, read_only=True)
-    cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    cinema_hall_name = serializers.CharField(
+        source="cinema_hall.name", read_only=True
+    )
+    cinema_hall_capacity = serializers.IntegerField(
+        source="cinema_hall.capacity", read_only=True
+    )
+    movie_title = serializers.CharField(source="movie.title", read_only=True)
+    movie = MovieSerializer(read_only=True)
+    cinema_hall = CinemaHallSerializer(read_only=True)
+    taken_places = serializers.SerializerMethodField()
```

Comment:
Do not overcomplicate this serializer, leave previous fields as is and create a new serializer for `taken_places`
```suggestion
    movie = MovieListSerializer(many=False, read_only=True)
    cinema_hall = CinemaHallSerializer(many=False, read_only=True)
    taken_places = TicketTakenSeatsSerializer(many=True, source="tickets")
```